docs/charmed_k8s_pod.md

```diff
@@ -0,0 +1,87 @@
+## Working with Charmed Kubernetes From within a pod
+
+### Setup
+
+After installing [Juju](https://juju.is/docs/olm/install-juju) and [Charmed Kubernetes](https://ubuntu.com/kubernetes/docs/install-manual) and [setting it up](https://ubuntu.com/kubernetes/docs/operations), let's discuss the steps to launch Spark jobs from within a pod in Charmed Kubernetes.
+
+First thing, we launch a pod using Canonical's Charmed Spark container image.
+
+Create a pod manifest ```shell-demo.yaml``` for use with Charmed Kubernetes, something like this
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: shell-demo
+  namespace: spark-test-ns
+spec:
+  containers:
+  - name: spark-client
+    image: dataplatformoci/spark:3.3.2
+    command: ["/bin/pebble", "run", "--hold"]
+  serviceAccountName: spark-user
+  hostNetwork: true
+  dnsPolicy: Default
+```
+
+The pod can be created with
+```shell
+kubectl apply -f shell-demo.yaml
+```
+
+after that, one logs in with
+
+```shell
+$ kubectl exec --stdin --tty shell-demo -n spark-test-ns -- /bin/bash 
+```
+
+Now from within the pod, lets create the kubeconfig for use with spark-client. ```KUBECONFIG_CONTENTS_FROM_CHARMED_KUBERNETES``` will come from the first step executed outside the Charmed Kubernetes cluster pod. 
```

Comment:
Are you sure this step is needed? I think the necessary information should [all be available](https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#directly-accessing-the-rest-api) to the container already or can be set in additional [environment variables](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/) made available to the container.

Reply:
Thanks for pointing out Rob! 

Our spark-client needs to be updated to be able to work with this information within the pod instead of insisting for a kubeconfig file which is the case right now.

@deusebio, @grobbie - I plan to file this and address this separately. Let me know your thoughts.

Reply:
Yes, best to have a ticket for this since it has impacts in the code (which is outside the scope of this PR)

Reply:
Created [DPE-1491](https://warthogs.atlassian.net/browse/DPE-1491) to address this requirement.